### PR TITLE
feat: added ape cache plugin

### PR DESCRIPTION
**Pull request by johnson2427**
### What I did

Moved DefaultQueryProvider to ape cache
Created initial database structure
added pydantic schemas

fixes: #632

### How I did it
Moved DefaultQueryProvider to ape cache
created cli and added init, purge and query functions

### How to verify it
```bash
ape cache init
ape console
```
```python
for _ in range(10000):
   chain.mine()
chain.blocks.query("number")
```

```bash
ape cache query "SELECT COUNT(*)  FROM blocks"
```

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated
- [ ] Fix commits (26 ahead of ape)

### CODE REVIEW

**Comment by johnson2427** [comment]
ignore my screw up with the merge

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ repos:
     -   id: flake8
 
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.950
+    rev: v0.942
```
Comment (unparalleled-js) [inline .pre-commit-config.yaml:24]: This should be `v0.950` though?

File: README.md [hunk 2]
```diff
@@ -1,70 +1,26 @@
-# Quick Start
```
Comment (unparalleled-js) [inline README.md:1]: Are you accidentally undo-ing @Ninjagod1251 's changes here?
Are you not caught up with main?

**Review by unparalleled-js — COMMENTED** [review]
Will look more when the diff is straightened out!

File: README.md (hunk 2 above)
Comment (fubuloubu) [reply README.md:1]: Yeah, has to rebase on main

File: .pre-commit-config.yaml (hunk 1 above)
Comment (johnson2427) [reply .pre-commit-config.yaml:24]: let me back track, my pycharm didn't grab the changes properly and sent me down a deep rabbit hole

File: README.md (hunk 2 above)
Comment (johnson2427) [reply README.md:1]: Yeah, I'm not sure what happened when I went to rebase, but it didn't show me the differences like it should have and showed them in reverse for some of them

**Comment by johnson2427** [comment]
Issue with rebase

File: README.md (hunk 2 above)
Comment (fubuloubu) [reply README.md:1]: > Yeah, I'm not sure what happened when I went to rebase, but it didn't show me the differences like it should have and showed them in reverse for some of them

Some of your commits are probably older then.

first make sure your version `main` is up to date with `upstream`, then checkout this branch again and do `git rebase -i main` and remove any commits that you didn't mean to add. finally, do `git push -f` if you are happy with the update only containing changes you've made based off of `main`

**Pull request closed without merging**